src/AzureFunctions/Properties.xml

```diff
@@ -0,0 +1,138 @@
+<?xml version="1.0"?>
+<Content xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Object id="AZURE_FUNCTIONS">
+    <Group>
+      <Id>Cloud</Id>
+      <Name>Cloud Settings</Name>
+      <Description>Settings for Azure Functions deployment</Description>
+      <Type>Main</Type>
+      <Childs>
+	      <Prop>
+          <Id>AZURE_FUNCTIONS_FUNCTION_APP</Id>
+          <Name>Function app</Name>
+          <Description>Name of the Function app in Azure. It must start with a letter and only contain letters, numbers, _, and -, up to 127 characters in length.</Description>
+          <Type>Text</Type>
+          <Default></Default>
+      </Prop>
+	    <Prop>
+          <Id>AZURE_FUNCTIONS_RESOURCE_GROUP</Id>
+          <Name>Resource group</Name>
+          <Description>Name of the resource group where the function app is defined in Azure.</Description>
+          <Type>Text</Type>
+          <Default></Default>
+      </Prop>
+	    <Group>
+      <Id>ServiceIdentityConfiguration</Id>
+      <Name>Identity Configuration</Name>
+      <Description>An Azure service principal is an identity created for use with applications, hosted services, and automated tools to access Azure resources.</Description>
+      <Type>Main</Type>
+      <Childs>
+        <Prop>
+          <Id>AZURE_FUNCTIONS_SP_APP_ID</Id>
+          <Name>Service Principal Application Id</Name>
+          <Description>Application (client) ID of the Service Principal.</Description>
+          <Type>Text</Type>
+          <Options>
+            <Option type='Required'/>
+          </Options>
+        </Prop>
+        <Prop>
+          <Id>AZURE_FUNCTIONS_SP_TENANT_ID</Id>
+          <Name>Service Principal Tenant Id</Name>
+            <Description>Directory (tenant) ID of the Service Principal.</Description>
+            <Type>Text</Type>
+            <Options>
+              <Option type='Required'/>
+            </Options>
+          </Prop>
+      <Prop>
+            <Id>AZURE_FUNCTIONS_SP_CREDENTIALS</Id>
```

Comment:
This is just a question. I'm not sure if this applies or not. 

I remember that in the Deploy To Azure Target, you only need to Specify a JSON file with all the credentials and configuration.  Please check Deploy to Azure Target.

Maybe this configuration is the same as the JSON config File of the Azure Deployment. 
Surely a JSON config file is more friendlier than 4 or 5 properties.

Reply:
Yes, it's the suggested way to authenticate using tools:
https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli

An Azure service principal is an identity created for use with applications, hosted services, and automated tools to access Azure resources. This access is restricted by the roles assigned to the service principal, giving you control over which resources can be accessed and at which level. For security reasons, it's always recommended to use service principals with automated tools rather than allowing them to log in with a user identity."